### ISSUE TITLE
Movement of wall charges sound inconsistent when using keyboard nav

### ISSUE THREAD

**Issue by Nancy-Salpepi** [phetsims/balloons-and-static-electricity#533]
**Test device**
MacBook Air (m1 chip)

**Operating System**
11.4

**Browser**
chrome

**Problem description**
https://github.com/phetsims/qa/issues/706

When a charged balloon is moved towards the wall at the bottom of the play area, the sound does not change pitch, which is what I hear when it is moved towards the wall from the top or center of the play area. 

**Steps to reproduce**
1. Use keyboard nav to move the yellow balloon on sweater to pick up charges
2. Move the balloon to the top of the play area, near the sweater 
3. Use shift + right/left arrows to slowly move the balloon across the top of the play area to the wall and then back again. 
4. Repeat step 3, but at the very bottom of the play area. 

You can also hear the difference in sound without using the shift key. I hear the pitch change when using the trackpad instead of keyboard navigation. 

**Visuals**

https://user-images.githubusercontent.com/87318828/132729175-4194a427-4e4e-44dd-b5b7-0ca9dba14714.mov

Troubleshooting information:
!!!!! DO NOT EDIT !!!!!
Name: ‪Balloons and Static Electricity‬
URL: https://phet-dev.colorado.edu/html/balloons-and-static-electricity/1.5.0-rc.1/phet/balloons-and-static-electricity_all_phet.html
Version: 1.5.0-rc.1 2021-09-07 23:14:39 UTC
Features missing: applicationcache, applicationcache, touch
Flags: pixelRatioScaling
User Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/93.0.4577.63 Safari/537.36
Language: en-US
Window: 1351x659
Pixel Ratio: 2/1
WebGL: WebGL 1.0 (OpenGL ES 2.0 Chromium)
GLSL: WebGL GLSL ES 1.0 (OpenGL ES GLSL ES 1.0 Chromium)
Vendor: WebKit (WebKit WebGL)
Vertex: attribs: 16 varying: 31 uniform: 1024
Texture: size: 16384 imageUnits: 16 (vertex: 16, combined: 80)
Max viewport: 16384x16384
OES_texture_float: true
Dependencies JSON: {}

**Comment by jbphet** [comment]
Another good catch @Nancy-Salpepi - thanks.

This problem occurs because there is code that limits the number of sound events that are played during keyboard dragging because there are a ton of such dragging events that occur, and playing sounds for all of them sounds chaotic.  The code to limit the number of plays was using a list of the sound generators that was ordered from top to bottom with respect to the charges with which they were associated.  This led to a case where the top sound generator was getting preferential treatment, meaning that it was being played more often.  Thus, for balloons at the bottom, the sounds for the charges closest to them were being played less frequently, leading to a set of sounds that weren't changing pitch.

My proposed solution to this is to order the sound generators based on proximity to the balloon when playing them.  Then, if some aren't played, it will be the ones that are furthest from the balloon.  This seems to lead to more consistent behavior.

I'll review the behavior with the sound design team in today's meeting and, if they're good with it, we'll call this fixed.

**Comment by Nancy-Salpepi** [comment]
Thanks @jbphet

I also noticed that when using the shortcut keys J+W, the only sound made is for the balloon hitting the wall. There is no sound for the movement of the electrons. Is that acceptable behavior or shall I open a separate issue for this?

**Comment by jbphet** [comment]
The sound design team listened to the changes for keyboard drag and are good with it.

@Nancy-Salpepi asked:
 > I also noticed that when using the shortcut keys J+W, the only sound made is for the balloon hitting the wall. There is no sound for the movement of the electrons. Is that acceptable behavior or shall I open a separate issue for this?

@emily-phet said that using the shortcut keys is seen as a way to instantaneously set a state, so not producing sounds that result from this state change is acceptable.  In other words, it's fine as is.
